src/dotnet/SlnFileFactory.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.DotNet.Cli.Sln.Internal;
+using Microsoft.DotNet.Cli.Utils;
+
+namespace Microsoft.DotNet.Tools.Common
+{
+    public static class SlnFileFactory
+    {
+        public static SlnFile CreateFromFileOrDirectory(string fileOrDirectory)
+        {
+            if (File.Exists(fileOrDirectory))
+            {
+                return FromFile(fileOrDirectory);
+            }
+            else
+            {
+                return FromDirectory(fileOrDirectory);
+            }
+        }
+
+        private static SlnFile FromFile(string solutionPath)
+        {
+            SlnFile slnFile = null;
+            try
+            {
+                slnFile = SlnFile.Read(solutionPath);
+            }
+            catch
+            {
```

Comment:
What kinds of errors are we expecting here? It looks like `SlnFile.Read` could potentially throw a few different types of exceptions, not all of which would be accurately described by `CommonLocalizableStrings.InvalidSolution`.